src/runner.ts

```diff
@@ -477,11 +502,48 @@ export class GitHubRunners extends Construct {
     this.secrets.setup.grantRead(setupFunction);
     this.secrets.setup.grantWrite(setupFunction);
 
-    return setupFunction.addFunctionUrl({ authType: FunctionUrlAuthType.NONE }).url;
+    if (access === LambdaAccessType?.LAMBDA_URL) {
+      return setupFunction.addFunctionUrl({ authType: FunctionUrlAuthType.NONE }).url;
+    } else if (access === LambdaAccessType?.API_GATEWAY) {
+      const api = new apigateway.LambdaRestApi(this, 'api', {
+        handler: setupFunction,
+        proxy: false,
+        cloudWatchRole: false,
+        policy: PolicyDocument.fromJson({
+          Version: '2012-10-17',
+          Statement: [
+            {
+              Effect: 'Allow',
+              Principal: '*',
+              Action: 'execute-api:Invoke',
+              Resource: 'execute-api:/*/*/*',
+              Condition: {
```

Comment:
would be great to have this configurable ... e.g. for Github Enterprise Server this won`t work.